### PR TITLE
Qwebengine fix

### PR DESCRIPTION
**Pull request by zocker-160**
As discussed on Reddit, this is a complete rewrite from scratch of all manifests.
It uses the latest version of PyQt5 and fixes the QWebengine problem.

This is a draft for now so I can do some testing with the version build by the flathub-bot.

### CODE REVIEW

**Comment by hfiguiere** [comment]
changing the format of the manifest is the best way to obfuscate the changes. Is this really necessary?

**Comment by hfiguiere** [comment]
also "python3-modules.json" was likely autogenerate by the flatpak-builder-tools, making the use of the tool impossible down the way, making the maintainer life harder.

File: ch.theologeek.Manuskript.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  - --device=dri
+  - --filesystem=home
+  #- --env=QT_QPA_PLATFORM=xcb
+  #- --env=TMPDIR=/var/tmp
+  # add paths of TeXlive Flatpak extension binaries
+  - --env=PATH=/usr/bin:/app/bin:/app/texlive/bin:/app/texlive/bin/x86_64-linux:/app/texlive/bin/aarch64-linux
+  - --env=QTWEBENGINEPROCESS_PATH=/app/bin/QtWebEngineProcess # allow using the QtWebEngine
+
+cleanup:
+  - /lib/debug
```
Comment (hfiguiere) [inline ch.theologeek.Manuskript.yml:29]:
```suggestion
```

don't cleanup these

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by zocker-160** [comment]
> changing the format of the manifest is the best way to obfuscate the changes. Is this really necessary?

This is a complete rewrite, nothing to be obfuscated here. YML is the easier format to work with, that is why I use it.

regenerating the autogenerated file is not a problem

File: ch.theologeek.Manuskript.yml (hunk 1 above)
Comment (zocker-160) [reply ch.theologeek.Manuskript.yml:29]: why? I decreases the size of the Flatpak by 80MB

Comment (hfiguiere) [reply ch.theologeek.Manuskript.yml:29]: not sure how you check. /lib/debug is separated by flatpak builder and put into it's own `.Debug` package

**Comment by zocker-160** [comment]
the autogenerated module is back to json now

File: ch.theologeek.Manuskript.yml (hunk 1 above)
Comment (zocker-160) [reply ch.theologeek.Manuskript.yml]: I don't know how that would be useful, but I will keep it then.

File: ch.theologeek.Manuskript.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    modules:
+      - name: texlive-extension
+        buildsystem: simple
+        build-commands:
+          - mkdir /app/texlive
+
+  - name: manuskript
+    buildsystem: simple
+    build-commands:
+      - install -d -m755 manuskript /app/lib/manuskript
```
Comment (hfiguiere) [inline ch.theologeek.Manuskript.yml:76]:
```suggestion
      - install -d  /app/lib/manuskript
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by zocker-160** [comment]
I guess I need to convert this to an actual PR in order to get the flathubbot building?

**Comment by hfiguiere** [comment]
> I guess I need to convert this to an actual PR in order to get the flathubbot building?

no, it should have been detected


bot, build

**Comment by TheJackiMonster** [comment]
Okay, so I tested the build and it seems to work as intended. The PDF preview finally works which is awesome. The installation gets cleaned up a bit now. The only downside would be that the build for aarch64 fails.

The log states that it mostly fails because of the missing type `QOpenGLTimerQuery`. So I would think enabling QtOpenGL for PyQt5 might fix that.

**Comment by zocker-160** [comment]
> The only downside would be that the build for aarch64 fails.

Worst case would be to disable the pdf preview for aarch64.

I wonder why the build failed this time though, it looks like some network problem, since it did not even start to build.

bot, build

**Comment by zocker-160** [comment]
looks like the same error is happening again :(

**Comment by hfiguiere** [comment]
You need to disable desktopGL on aarch64. Because the SDK is built without it on aarch64.

It worked with the previous manifest though.

**Comment by zocker-160** [comment]
bot, build

**Comment by zocker-160** [comment]
I don't quite understand what the error is, but it is happening during the build of QtGui, so I am not sure if this is even related to QtOpenGL

**Comment by zocker-160** [comment]
@TheJackiMonster alright so building all PyQt modules (like the old manifest did) seems to solve the build issues on aarch64.

**Comment by zocker-160** [comment]
bot, build

**Comment by zocker-160** [comment]
bot, build

**Comment by zocker-160** [comment]
nice it seems to work! Please review @TheJackiMonster :)

**Review by TheJackiMonster — APPROVED** [review]
Looks good to me. I will probably add the import into the startup script from the repository or maybe even into the script using the QtWebWidgets. Then I should replace the `manuskript.py` with the usual startup script from repository again.

But the import in the added `manuskript.py` seems to be necessary for the PDF preview to work. For now this PR seems to be a good solution. I will make the required adjustments with the next release but these changes should improve the current flatpak already for version 0.13.1 a lot. ^^

Thank you very much.

**Pull request merged**